static/app/components/globalModal.tsx

```diff
@@ -1,97 +1,276 @@
+  const closeModal = React.useCallback(() => {
+    // Option close callback, from the thing which opened the modal
+    options.onClose?.();
+
+    // Action creator, actually closes the modal
+    actionCloseModal();
+
+    // GlobalModal onClose prop callback
+    onClose?.();
+  }, [options]);
+
+  const handleEscapeClose = React.useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        closeModal();
+      }
+    },
+    [closeModal]
+  );
+
+  const portal = getModalPortal();
+  const focusTrap = React.useRef<FocusTrap>();
+
+  React.useEffect(() => void (focusTrap.current = createFocusTrap(portal)), []);
+
+  React.useEffect(() => {
+    const body = document.querySelector('body');
+    const root = document.getElementById(ROOT_ELEMENT);
+
+    if (!body || !root) {
+      return () => void 0;
     }
-  };
 
-  render() {
-    const {visible, children, options} = this.props;
-    const renderedChild =
-      typeof children === 'function'
-        ? children({
-            closeModal: this.handleCloseModal,
-            Header: Modal.Header,
-            Body: Modal.Body,
-            Footer: Modal.Footer,
-          })
-        : undefined;
-
-    if (options && options.type === 'confirm') {
-      return <Confirm onConfirm={() => {}}>{() => renderedChild}</Confirm>;
+    const reset = () => {
+      body.style.removeProperty('overflow');
+      root.removeAttribute('aria-hidden');
+      focusTrap.current?.deactivate();
+      portal.removeEventListener('keydown', handleEscapeClose);
+    };
+
+    if (visible) {
+      body.style.overflow = 'hidden';
+      root.setAttribute('aria-hidden', 'true');
+      focusTrap.current?.activate();
+      portal.addEventListener('keydown', handleEscapeClose);
+    } else {
+      reset();
     }
 
-    return (
-      <ClassNames>
-        {({css, cx}) => (
-          <Modal
-            className={cx(
-              options?.modalClassName,
-              options?.modalCss && css(options.modalCss)
-            )}
-            dialogClassName={options && options.dialogClassName}
-            show={visible}
-            animation={false}
-            onHide={this.handleCloseModal}
-            backdrop={options?.backdrop}
-          >
-            {renderedChild}
-          </Modal>
-        )}
-      </ClassNames>
-    );
-  }
+    return reset;
+  }, [visible]);
+
+  const Header = makeClosableHeader(closeModal);
+  const renderedChild = children?.({Header, Body, Footer, closeModal});
+
+  // Default to enabled backdrop
+  const backdrop = options.backdrop ?? true;
+
+  return ReactDOM.createPortal(
+    <React.Fragment>
+      <Backdrop
+        onClick={backdrop === true ? closeModal : undefined}
+        style={backdrop && visible ? {opacity: 0.5, pointerEvents: 'auto'} : {}}
+      />
+      <Container>
+        <AnimatePresence>
+          {visible && (
+            <Modal role="dialog" className="modal-dialog" css={options.modalCss}>
```

Comment:
do we still need the `modal-dialog` class?

Reply:
Got rid of it!